src/main/java/lotto/domain/Lotto.java

```diff
@@ -0,0 +1,49 @@
+package lotto.domain;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+public class Lotto {
+
+    private static final int START_RANGE = 1;
+    private static final int END_RANGE = 45;
+    private static final int COUNT = 6;
+    private final List<Integer> lottoNumbers;
+
+    public Lotto() {
+        this.lottoNumbers = generateRandomNumbers();
+    }
+
+    public Lotto(List<Integer> lottoNumbers) {
+        this.lottoNumbers = lottoNumbers;
+    }
+
+    private List<Integer> generateRandomNumbers() {
+        List<Integer> nums = IntStream.rangeClosed(START_RANGE, END_RANGE)// IntStream
+            .boxed() // Stream<Integer>
+            .collect(Collectors.toList()); // List<Integer>
+        Collections.shuffle(nums);
+
+        List<Integer> newNums = nums.subList(0, COUNT);
+        Collections.sort(newNums);
+
+        return newNums;
+    }
+
+    public int compareLottoAndWinnerNumbers(LottoMachine lottoMachine) {
+        return Long.valueOf(lottoNumbers.stream()
+            .filter(lottoMachine::isContain).count()).intValue();
+    }
+
+    public int compareLottoAndBonusNumbers(final LottoMachine lottoMachine) {
+        return Long.valueOf(lottoNumbers.stream()
+            .filter(lottoMachine::isEqualBonusNumber).count()).intValue();
+    }
```

Comment:
아래와 같이 long -> Long -> int로 변경할 필요 없이 `명시적 형변환`으로 변경해볼 수 있을 것 같아요. 🤗
```suggestion
    public int compareLottoAndWinnerNumbers(LottoMachine lottoMachine) {
        return (int) lottoNumbers.stream()
                .filter(lottoMachine::isContain)
                .count();
    }

    public int compareLottoAndBonusNumbers(final LottoMachine lottoMachine) {
        return (int) lottoNumbers.stream()
                .filter(lottoMachine::isEqualBonusNumber)
                .count();
    }
```

이 두 메서드는 `Lotto` 객체가 아니라 LottoMachine 객체의 역할이 아닐까요? 🤔